unstable/wlr-surface-layers-unstable-v1.xml

```diff
@@ -0,0 +1,231 @@
+      atomically applies all pending state, replacing the current state. After
+      commit, the new pending state is as document for each related request.
+    </description>
+
+    <request name="set_interactivity">
+      <description summary="indicates that this surface is interactive">
+        This request indicates that the surface would like to receive input
+        events when focused. The precise semantics of focus are
+        compositor-defined.
+
+        Interactivity is double-buffered, see wl_surface.commit.
+      </description>
+      <arg name="interactive" type="bool" />
+    </request>
+
+    <request name="set_anchor">
+      <description summary="configures the anchor point of the surface">
+        Requests that the compositor anchor the surface to the specified edges
+        and corners. If two orthoginal edges are specified (e.g. 'top' and
+        'left'), then the anchor point will be the intersection of the edges
+        (e.g. the top left corner of the output); otherwise the anchor point
+        will be centered on that edge, or in the center if none is specified.
+
+        Anchor is double-buffered, see wl_surface.commit.
+      </description>
+      <arg name="anchor" type="uint" enum="anchor"/>
+    </request>
+
+    <request name="set_exclusive_zone">
+      <description summary="configures the exclusive geometry of this surface">
+        Requests that the compositor avoids occluding an area of the surface
+        with other surfaces. The compositor's use of this information is
+        implementation-dependent - do not assume that this region will not
+        actually be occluded.
+
+        This value is only meaningful if the surface is anchored to an edge,
+        rather than a corner. The zone is the number of pixels from the edge
+        that are considered exclusive.
+
+        Exclusive zone is double-buffered, see wl_surface.commit.
+      </description>
+      <arg name="zone" type="uint"/>
+    </request>
+
+    <request name="set_margin">
+      <description summary="sets a margin from the anchor point">
+        Requests that the surface be placed some distance away from the anchor
+        point on the output, in pixels. Setting this value for edges you are
+        not anchored to has no effect.
+
+        Margin is double-buffered, see wl_surface.commit.
+      </description>
+      <arg name="top" type="int"/>
+      <arg name="right" type="int"/>
+      <arg name="bottom" type="int"/>
+      <arg name="left" type="int"/>
+    </request>
+
+    <request name="get_popup">
+      <description summary="assign this layer_surface as an xdg_popup parent">
+        This assigns an xdg_popup's parent to this layer_surface.  This popup
+        should have been created via xdg_surface::get_popup with the parent set
+        to NULL, and this request must be invoked before committing the popup's
+        initial state.
+
```

Comment:
https://github.com/wayland-project/wayland-protocols/blob/master/stable/xdg-shell/xdg-shell.xml#L1010
mandates that the parent  of a xdg_popup needs to be a xdg_toplevel or xdg_popup surface so strictly speaking using a layer_surface would be a violation of the xdg protocol. I wonder if it might confuser toolkits somehow if this is violated.

Reply:
You're right. I think the NULL parent in `get_popup` was originally designed for xdg-foreign.

To fix this, we could duplicate `xdg_popup` into `wlr_layer_popup` and keep using `xdg_positioner`.

Reply:
>To fix this, we could duplicate xdg_popup into wlr_layer_popup and keep using xdg_positioner.

This is probably appropriate. But we should float a discussion on the mailing list for removing this restriction from xdg-shell, it wouldn't be a breaking change.